### PR TITLE
python312Packages.phunspell: init at 0.1.6

### PR DESCRIPTION
**Pull request by ViZiD**
## Description of changes
Pure Python spell checker, wrapping spylls a port of Hunspell
https://github.com/dvwright/phunspell

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/phunspell/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-TlO9Ccr2iRN+s3JV+3P36RF9oFY32fj+24sKslZZCVk=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "fix-deprecation-warnings.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/dvwright/phunspell/pull/2.patch";
+      hash = "sha256-uEgMKqtvBn+/xE7Qp2sMpoADm0+NnuCDKA8UEBo/2uA=";
+    })
```
Comment (getchoo) [inline pkgs/development/python-modules/phunspell/default.nix:29]:
```suggestion
    (fetchpatch {
      name = "fix-package-data-warning.patch";
      url = "https://github.com/dvwright/phunspell/commit/70a0ee8af8442797e03916cea31637c21e6d32d9.patch";
      hash = "sha256-qBVZAHlM4JUXbSYX1bgLFlMmLcK1V6K+lrVthIzGYvU=";
    })
    (fetchpatch {
      name = "replace-deprecated-option.patch";
      url = "https://github.com/dvwright/phunspell/commit/331c593b486cebe1a9b72befa568de9b51033f15.patch";
      hash = "sha256-CUBtDmRV4Da5i+DpeQnOaPXhUN1DBGxUVyRg7bX+Jj8=";
    })
```

using the commits directly here would be safer, as the content of `#2` could change at a later time

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/phunspell/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [ spylls ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/phunspell/default.nix:35]: Could you add `pythonImportsCheck` for a simple check?

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 308344` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.phunspell</li>
    <li>python311Packages.phunspell.dist</li>
    <li>python312Packages.phunspell</li>
    <li>python312Packages.phunspell.dist</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**